Author: Andreas Beckmann <debian@abeckmann.de>
Description: Module build fix for 2.6.36
 In Linux 2.6.36 the member 'ioctl' of struct file_operations was removed,
 resulting in the following build failure:
     nv.c:426: error: unknown field ‘ioctl’ specified in initializer
 This patch adds some guards around its usage.
 A replacement (unlocked_ioctl) was already introduced by NVIDIA.

Index: b/nv.c
===================================================================
--- a/nv.c	2010-07-15 16:45:30.000000000 +0000
+++ b/nv.c	2010-09-26 07:28:38.580536890 +0000
@@ -322,7 +322,9 @@
 static struct file_operations nv_fops = {
     .owner     = THIS_MODULE,
     .poll      = nv_kern_poll,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,36)
     .ioctl     = nv_kern_ioctl,
+#endif
 #if defined(NVCPU_X86_64) && defined(HAVE_COMPAT_IOCTL)
     .compat_ioctl = nv_kern_compat_ioctl,
 #endif
